rllab/envs/dmcontrol_env.py

```diff
@@ -0,0 +1,130 @@
+import pygame
+import numpy as np
+from PIL import Image
+from dm_control import suite
+from dm_control.rl.environment import StepType
+from dm_control.suite.wrappers import pixels
+from rllab.envs.base import Env, Step
+from rllab.core.serializable import Serializable
+from rllab.spaces.box import Box
+from rllab.spaces.discrete import Discrete
+
+
+'''
+This environment will use dm_control toolkit to train and simulate your model.
+Domain name and task name used for dm_control is showed below: (domain name/task name)
+
+ball_in_cup/catch
+cartpole/swingup_sparse
+cartpole/balance_sparse
+cartpole/swingup
+cartpole/balance
+cheetah/run
+finger/turn_hard
+finger/spin
+finger/turn_easy
+fish/upright
+fish/swim
+hopper/stand
+hopper/hop
+humanoid/stand
+humanoid/run
+humanoid/walk
+manipulator/bring_ball
+pendulum/swingup
+point_mass/easy
+reacher/hard
+reacher/easy
+swimmer/swimmer6
+swimmer/swimmer15
+walker/stand
+walker/run
+walker/walk
+'''
+
+
+def flat_shape(observation):
+    dim = 0
+    for key, value in observation.items():
+        dim += int(np.prod(value.shape))
+    return dim
+
+def flat_observation(observation):
+    new_observation=[]
+    for key, value in observation.items():
```

Comment:
observation is a dict so it can return items() in any order it wants. the observations need to be stacked in the same order every time. sort by the key names first.

also, this could be made simpler with
```python
values = # get and sort values
return np.hstack(v.flatten() for v in values)
```

Reply:
The observation is a collections.OrderedDict, so it has been sorted.

Reply:
oh, good catch.